### PR TITLE
Add Bazel Docs

### PR DESCRIPTION
**Pull request by srikrsna-buf**

### CODE REVIEW

File: docs/breaking/usage.md [hunk 1]
```diff
@@ -156,9 +156,9 @@ google/type/date.proto:50:3:Field "3" on message "Date" changed type from "int32
 
 By default, `buf` builds all files under the `buf.yaml` configuration file. You can instead
 manually specify the file or directory paths to run breaking change detection. This is an advanced
-feature intended to be used for editor or Bazel integration - it is better to let `buf` discover
-all files under management and handle this for you in general, especially when using the `FILE`
-category.
+feature intended to be used for editor or [Bazel](/build-systems/bazel.md) integration - it is better
```
[the remark this answers is not in the document]
Comment (srikrsna-buf) [reply docs/breaking/usage.md:159]: Yes, it works in [preview](https://deploy-preview-160--buf.netlify.app/breaking/usage#limit-to-specific-files)

File: docs/build-systems/bazel.md [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+
+# Protobuf rules
+load("@rules_proto//proto:repositories.bzl", "rules_proto_dependencies", "rules_proto_toolchains")
+
+rules_proto_dependencies()
+
+rules_proto_toolchains()
+```
+
+### Using a specific version of the Protobuf rules
```
Comment (bufdev) [inline docs/build-systems/bazel.md:44]: You likely want to change this from "Protobuf rules" to "rules_proto", I read this as setting a specific version of "rules_buf", which I thought was covered in the previous section.

File: docs/build-systems/bazel.md [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+
+This can be run as:
+
+```sh
+$ bazel test :foo_proto_lint
+```
+
+We recommend having a single `buf_lint_test` for each `proto_library` target. The [Gazelle extension](#gazelle) can generate them in the same pattern.
+
+#### Attributes
```
Comment (bufdev) [inline docs/build-systems/bazel.md:127]: I would probably put attributes before example

File: docs/build-systems/bazel.md [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+
+Name  | Description | Type | Mandatory | Default
+:-----| :-----------| :--- | :-------- | :------
+`name` |  A unique name for this target.   | [Name][config_name] | required |  |
+`config` |  The [`buf.yaml`][buf_yaml] file. | [Label][config_label] | optional | None |
+`targets` |  `proto_library` targets to lint | [List of labels][config_label] | required |  |
+
+### `buf_breaking_test` {#buf-breaking-test}
+
+`buf_breaking_test` is a test rule that checks one or more `proto_library` targets for breaking changes. It requires an [image](/reference/images) file to check against. Here's an example usage of the rule:
```
Comment (bufdev) [inline docs/build-systems/bazel.md:139]: We should discuss the requirement to have an against image.

File: docs/build-systems/bazel.md [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+
+[//]: # (The table is copied from documentation generated by stardoc: https://github.com/bufbuild/rules_buf/blob/main/buf/README.md )
+
+Name  | Description | Type | Mandatory | Default
+:-----| :-----------| :--- | :-------- | :------
+`name` |  A unique name for this target.   | [Name][config_name] | required |  |
+`config` |  The [`buf.yaml`][buf_yaml] file. | [Label][config_label] | optional | None |
+`targets` |  `proto_library` targets to lint | [List of labels][config_label] | required |  |
+
+### `buf_breaking_test` {#buf-breaking-test}
```
Comment (bufdev) [inline docs/build-systems/bazel.md:137]: `buf_lint_test` has `#### Example` but this section does not, we should be consistent

File: docs/build-systems/bazel.md [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+
+This can be run as:
+
+```terminal
+$ bazel test :foo_proto_breaking
+```
+
+We recommend having a single `buf_breaking_test` for each Buf module. For repositories that contain a `buf.work.yaml` that references multiple `buf.yaml` files, there needs to be exactly one `buf_breaking_test` for each `buf.yaml` file.
+
+Alternatively, a single `buf_breaking_test` can be used against each `proto_library` target. For this to work, `limit_to_input_files` attribute must be set to `True` as the against image file may contain other Protobuf files. Although this is closer to how bazel operates, for this particular use case it is not recommended. This [section](#example-module-vs-package-mode) explains the differences with an example.
```
Comment (robbertvanginkel) [inline docs/build-systems/bazel.md:166]:
```suggestion
Alternatively, a single `buf_breaking_test` can be used against each `proto_library` target. For this to work, `limit_to_input_files` attribute must be set to `True` as the against image file may contain other Protobuf files. Although this is closer to how bazel operates, for this particular use case it is not recommended. See the [module vs package mode example](#example-module-vs-package-mode) for a concrete example of the differences.
```

I'm generally not a fan of "see [this]" thing where the this doesnt really indicate where it goes.

**Review by robbertvanginkel — COMMENTED** [review]
Few small suggestions, mostly looks good to me

File: docs/build-systems/bazel.md [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+# Load the other rules_buf assets
+```
+
+### Pinning the `buf` cli version
+
+By default, `rules_buf_toolchains` loads the latest buf version. For hermetic builds pin the cli version using the version attribute.
+
+```python title="WORKSPACE" {1-2}
+-rules_buf_toolchains()
++rules_buf_toolchains(version = "v1.5.0")
```
[the remark this answers is not in the document]
Comment (srikrsna-buf) [reply docs/build-systems/bazel.md:87]: Any version of `rules_buf` can download any version of `buf`

File: docs/build-systems/bazel.md [hunk 8]
```diff
@@ -91,6 +91,44 @@ The rules work alongside `proto_library` rules. You can configure `rules_buf` us
 
 > We recommend using the [Gazelle extension](#gazelle) to generate the following rules.
 
+### `buf_dependencies` {#buf-dependencies}
+
+`buf_dependencies` is a repository rule that downloads one or more modules from the [BSR][bsr] and generates build files using Gazelle. [Setup Gazelle][gazelle_setup] to use this rule. To also use Gazelle to generate this rule see the [Dependencies](#gazelle-dependencies) section.
```
Comment (bufdev) [inline docs/build-systems/bazel.md:96]: What actually happens here? Are these vendored? How do they stay up to date? This should all be documented

Comment (srikrsna-buf) [reply docs/build-systems/bazel.md:96]: > Are these vendored? How do they stay up to date?

As a bazel user doesn't the repository rule part answer these?

File: docs/build-systems/bazel.md [hunk 9]
```diff
@@ -0,0 +1,491 @@
...
+)
+```
+
+This can be run as:
+
+```terminal
+$ bazel test :foo_proto_breaking
+```
+
+We recommend having a single `buf_breaking_test` for each Buf module. For repositories that contain a `buf.work.yaml` that references multiple `buf.yaml` files, there needs to be exactly one `buf_breaking_test` for each `buf.yaml` file.
```
Comment (amckinney) [inline docs/build-systems/bazel.md:214]: Since we reference the `buf.yaml` immediately after, this might be more clear if we drop "Buf module" and replace it with `buf.yaml`:

```suggestion
We recommend having a single `buf_breaking_test` for each `buf.yaml`. For repositories that contain a `buf.work.yaml` that references multiple `buf.yaml` files, there needs to be exactly one `buf_breaking_test` for each `buf.yaml` file.
```

File: docs/build-systems/bazel.md [hunk 10]
```diff
@@ -0,0 +1,491 @@
...
+
+#### Against image
+
+You can generate an [Image] file like this:
+
+```terminal
+$ buf build --exclude-imports -o image.bin <input>
+```
+
+The `<input>` can be a path to a Buf module or in one of [several other Image formats](/reference/inputs).
```
Comment (amckinney) [inline docs/build-systems/bazel.md:228]: Given that we're talking about generic inputs here, we could rephrase this a bit to make it a little more clear. Something along the lines of the following:

```suggestion
The `<input>` is often a directory containing a `buf.yaml`, but all of the other [Input formats](/reference/inputs) are also supported.
```

**Review by amckinney — COMMENTED** [review]
This looks great overall - I have several suggestions for adding some more clarity in certain places. Aside from that let's just make sure there's enough example material, and we should be good to go!

File: docs/build-systems/bazel.md [hunk 11]
```diff
@@ -0,0 +1,491 @@
...
+
+## Rules
+
+The rules work alongside `proto_library` rules. You can configure `rules_buf` using a [`buf.yaml`][buf_yaml] configuration file. Export the `buf.yaml` using `exports_files(["buf.yaml"])` to reference it. For repositories that contain a [`buf.work.yaml`][buf_work_yaml] that references multiple `buf.yaml` files, you need to export and reference each `buf.yaml` file independently.
+
+> We recommend using the [Gazelle extension](#gazelle) to generate the following rules.
+
+### `buf_dependencies` {#buf-dependencies}
+
+`buf_dependencies` is a repository rule that downloads one or more modules from the [BSR][bsr] and generates build files using Gazelle. [Setup Gazelle][gazelle_setup] to use this rule. To also use Gazelle to generate this rule see the [Dependencies](#gazelle-dependencies) section.
```
[the remark this answers is not in the document]
Comment (srikrsna-buf) [reply docs/build-systems/bazel.md:96]: The later part is true though, to hand write and use the rule you need gazelle because the rule runs `gazelle` to generate `proto_library` targets.
They can also use `gazelle` to generate this rule.

File: docs/build-systems/bazel.md [hunk 12]
```diff
@@ -0,0 +1,491 @@
...
+
+This will create the file `buf_deps.bzl` with the `buf_deps` macro that loads the `buf_dependencies` rules. It will also call the macro from the `WORKSPACE` file.
+
+#### Arguments
+
+Argument  | Mandatory | Default
+:-----|:--- | :--- 
+`-from_file file` <br/><br/> Must be one of `buf.work.yaml`, `buf.yaml`, `buf.lock`. When using `buf.work.yaml` or `buf.yaml` the rule will import from the associated `buf.lock` file(s). | required |  
+`-to_macro macroFile%defName` <br/><br/> Tells Gazelle to write new repository rules into a `.bzl` macro function rather than the `WORKSPACE` file. | optional | 
+`-prune true&#124;false` <br/><br/> When true, Gazelle will remove `buf_dependencies` rules that no longer have equivalent `buf.yaml` files. | optional | `false`
```
[the remark this answers is not in the document]
Comment (srikrsna-buf) [reply docs/build-systems/bazel.md:361]: `|`

File: docs/build-systems/bazel.md [hunk 13]
```diff
@@ -0,0 +1,491 @@
...
+
+```python title="WORKSPACE" {1-2}
+# rules_buf fetches the sha based on the version number, the version is enough for hermetic builds.
+-rules_buf_toolchains()
++rules_buf_toolchains(version = "v1.5.0")
+```
+
+## Rules
+
+The rules work alongside `proto_library` rules. You can configure `rules_buf` using a [`buf.yaml`][buf_yaml] configuration file. Export the `buf.yaml` using `exports_files(["buf.yaml"])` to reference it. For repositories that contain a [`buf.work.yaml`][buf_work_yaml] that references multiple `buf.yaml` files, you need to export and reference each `buf.yaml` file independently.
```
[the remark this answers is not in the document]
Comment (srikrsna-buf) [reply docs/build-systems/bazel.md:87]: I think we should add a link to `export_files` docs and we can also point them to the [examples](https://github.com/bufbuild/rules_buf/tree/main/examples) in the repo

File: docs/build-systems/bazel.md [hunk 14]
```diff
@@ -0,0 +1,489 @@
...
+load("@rules_proto//proto:repositories.bzl", "rules_proto_dependencies", "rules_proto_toolchains")
+
+rules_proto_dependencies()
+
+rules_proto_toolchains()
+```
+
+### Using a specific version of the `rules_proto`
+
+[`rules_proto`][rules_proto] is required to use `rules_buf`. By default, `rules_buf` automatically loads `rules_proto`, but you can use a specific version of it by loading it _before_ `rules_buf`. Here's an example:
```
Comment (smaye81) [inline docs/build-systems/bazel.md:43]: Might want to reword this to say:

> By default, `rules_buf` automatically loads the `rules_proto` `http_archive` so you don't need to specify that in your WORKSPACE file.  However, you can use a specific version of `rules_proto` if you like by loading the `http_archive` explicitly _before_ `rules_buf`.  Here's an example:

File: docs/build-systems/bazel.md [hunk 15]
```diff
@@ -0,0 +1,489 @@
...
+Alternatively, a single `buf_breaking_test` can be used against each `proto_library` target. For this to work, `limit_to_input_files` attribute must be set to `True` as the against image file may contain other Protobuf files. Although this is closer to how bazel operates, for this particular use case it is not recommended. See the [module vs package mode example](#example-module-vs-package-mode) for a concrete example of the differences.
+
+The [Gazelle extension](#gazelle) can generate `buf_breaking_test` in either levels of granularity.
+
+#### Image inputs {#image-input}
+
+You can generate an [Image][image] file like this:
+
+```terminal
+$ buf build --exclude-imports -o image.bin <input>
```
Comment (smaye81) [inline docs/build-systems/bazel.md:222]: Dumb nit, so feel free to ignore me, but maybe make this:

> The above rules can also use image files as inputs.  You can use the Buf CLI to generate an image file like this:

It was just kind of odd to me to see the CLI command here because all other commands seem to be Bazel-specific and then this was a one-off using the CLI.  It just makes it more clear to me /shrug.

File: docs/build-systems/bazel.md [hunk 16]
```diff
@@ -0,0 +1,489 @@
...
+
+This can be run as:
+
+```terminal
+$ bazel test :foo_proto_breaking
+```
+
+We recommend having a single `buf_breaking_test` for each `buf.yaml`. For repositories that contain a `buf.work.yaml` that references multiple `buf.yaml` files, there needs to be exactly one `buf_breaking_test` for each `buf.yaml` file.
+
+Alternatively, a single `buf_breaking_test` can be used against each `proto_library` target. For this to work, `limit_to_input_files` attribute must be set to `True` as the against image file may contain other Protobuf files. Although this is closer to how bazel operates, for this particular use case it is not recommended. See the [module vs package mode example](#example-module-vs-package-mode) for a concrete example of the differences.
```
Comment (smaye81) [inline docs/build-systems/bazel.md:213]: Maybe format against as `against` (second line 'must be set to `True` as the `against` image)

Comment (smaye81) [inline docs/build-systems/bazel.md:213]: Capitalize Bazel (second line 'closer to how Bazel operates')

File: docs/build-systems/bazel.md [hunk 17]
```diff
@@ -0,0 +1,489 @@
...
+The `<input>` is often a directory containing a `buf.yaml`, but all of the other [Input formats](/reference/inputs) are also supported.
+
+We recommend storing the Image file in a `testdata` directory and checking it in to version control and updating it as needed. In the case of repositories that follow a versioning scheme like [semver][semver], you can update it on each new release either manually or with a post-release hook.
+
+As an alternative to checking the Image file into version control, you can use CI artifacts. Many CI servers, like [Travis CI][travis], enable you to upload build artifacts to a backend like [S3][s3]. In CI, you can set up a pipeline to build the Image on each commit and then add those artifacts to your `WORKSPACE`:
+
+```python title="WORKSPACE"
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_file")
+
+# Assuming your using s3 and bucket is at http://s3-us-east-1.amazonaws.com/bucket/foo/bar
```
Comment (smaye81) [inline docs/build-systems/bazel.md:234]: `you're`

File: docs/build-systems/bazel.md [hunk 18]
```diff
@@ -0,0 +1,489 @@
...
+Export the `buf.yaml` file by adding `exports_files(["buf.yaml"])` to the `BUILD` file.
+
+> Inside Buf [workspaces][workspaces], make sure to export each `buf.yaml` file.
+
+Now run Gazelle
+```terminal
+$ bazel run //:gazelle
+```
+
+The buf gazelle extension will now take care of updating your proto build files, just run `//:gazelle` whenever proto files are added/removed.
```
Comment (smaye81) [inline docs/build-systems/bazel.md:320]: buf gazelle needs some sort of formatting or capitalization, but not sure what.

File: docs/build-systems/bazel.md [hunk 19]
```diff
@@ -0,0 +1,489 @@
...
+        "-to_macro=buf_deps.bzl%buf_deps",
+        # Deletes outdated repo rules
+        "-prune",
+    ],
+    command = "update-repos",
+    gazelle = ":gazelle-buf",
+)
+```
+
+Now run Gazelle update-repos command
```
Comment (smaye81) [inline docs/build-systems/bazel.md:345]: Now run the Gazelle `update-repos` command

File: docs/build-systems/bazel.md [hunk 20]
```diff
@@ -0,0 +1,489 @@
...
+```
+
+This will create the file `buf_deps.bzl` with the `buf_deps` macro that loads the `buf_dependencies` rules. It will also call the macro from the `WORKSPACE` file.
+
+#### Arguments
+
+Argument  | Mandatory | Default
+:-----|:--- | :--- 
+`-from_file file` <br/><br/> Must be one of `buf.work.yaml`, `buf.yaml`, `buf.lock`. When using `buf.work.yaml` or `buf.yaml` the rule will import from the associated `buf.lock` file(s). | required |  
+`-to_macro macroFile%defName` <br/><br/> Tells Gazelle to write new repository rules into a `.bzl` macro function rather than the `WORKSPACE` file. | optional |
```
Comment (smaye81) [inline docs/build-systems/bazel.md:357]: I'm assuming the default value for this is an empty string?  May want to specify that.

File: docs/build-systems/bazel.md [hunk 21]
```diff
@@ -0,0 +1,489 @@
...
+$ bazel run //:gazelle
+```
+
+Run this command to list the generated breaking rules:
+
+```terminal
+$ bazel query 'kind(buf_breaking_test, //...)'
+```
+
+This mimics running `buf breaking` on a module. This is the most accurate way to check for breaking changes. However, depending on multiple targets at once is an anti-pattern in bazel, so that's why we've provided package mode as an alternative.
```
Comment (smaye81) [inline docs/build-systems/bazel.md:400]: Capitalize Bazel

File: docs/build-systems/bazel.md [hunk 22]
```diff
@@ -0,0 +1,489 @@
...
+[//]: # (The table is copied from documentation generated by stardoc: https://github.com/bufbuild/rules_buf/blob/main/buf/README.md )
+
+Name  | Description | Type | Mandatory | Default
+:-----|:----------- | :--- | :-------- | :------
+`name` |  A unique name for this target.   | [Name][config_name] | required |  |
+`against` |  The image file to check against.  | [Label][config_label] | required |  |
+`config` |  The `buf.yaml` file.  | [Label][config_label] | optional | `None` |
+`exclude_imports` |  Exclude imports from breaking change detection.  | Boolean | optional | `False` |
+`limit_to_input_files` |  Only run breaking checks against the files in the targets. This has the effect of filtering the against image to only contain the files in the input. | Boolean | optional | `True` |
+`targets`|  `proto_library` targets to check for breaking changes | [List of labels][config_label] | `[]` |
```
Comment (smaye81) [inline docs/build-systems/bazel.md:184]: Is `targets` required or optional?  In the docs, the `[]` brackets show under the `Mandatory` header and `Defaults` is blank.

File: docs/build-systems/bazel.md [hunk 23]
```diff
@@ -0,0 +1,467 @@
...
+rules_proto_dependencies()
+
+rules_proto_toolchains()
+
+http_archive(
+    name = "rules_buf",
+    sha256 = "d6b2513456fe2229811da7eb67a444be7785f5323c6708b38d851d2b51e54d83",
+    urls = [        
+         "https://github.com/bufbuild/rules_buf/releases/download/v0.1.0/rules_buf-v0.1.0.zip",
+    ],
```
Comment (robbertvanginkel) [inline docs/build-systems/bazel.md:68]: Here we supply a sha256 and url, but line 21-25 have placeholders deferring to the readme. We should either have placeholders in both places, or keep the latest version around here and have the release instructions for rules_buf update these docs.

File: docs/build-systems/bazel.md [hunk 24]
```diff
@@ -0,0 +1,467 @@
...
+`buf_lint_test` is a test rule that lints one or more `proto_library` targets.
+
+#### Attributes
+
+[//]: # (The table is copied from documentation generated by stardoc: https://github.com/bufbuild/rules_buf/blob/main/buf/README.md )
+
+Name  | Description | Type | Mandatory | Default
+:-----| :-----------| :--- | :-------- | :------
+`name` |  A unique name for this target.   | [Name][config_name] | required |  |
+`config` |  The [`buf.yaml`][buf_yaml] file. | [Label][config_label] | optional | None |
```
Comment (robbertvanginkel) [inline docs/build-systems/bazel.md:139]: Lets note that if no config is specified, the default buf config will be applied (is that the same as you get from `buf mod init` or is there some other default?)

File: docs/build-systems/bazel.md [hunk 25]
```diff
@@ -0,0 +1,467 @@
...
+
+Run this command to list the generated lint rules:
+
+```terminal
+$ bazel query 'kind(buf_lint_test, //...)'
+```
+
+### Breaking change detection
+
+To run [breaking change detection][breaking] against Protobuf sources, you need to add a Gazelle [directive][gazelle_directive] that points to an [Image][image] target to generate breaking change detection rules. Gazelle directives are top-level commits in Bazel [`BUILD` files][build_files] that provide Gazelle with configuration.
```
Comment (robbertvanginkel) [inline docs/build-systems/bazel.md:346]:
```suggestion
To run [breaking change detection][breaking] against Protobuf sources, you need to add a Gazelle [directive][gazelle_directive] that points to an [Image][image] target to generate breaking change detection rules. Gazelle directives are top-level comments in Bazel [`BUILD` files][build_files] that provide Gazelle with configuration.
```

File: docs/build-systems/bazel.md [hunk 26]
```diff
@@ -0,0 +1,467 @@
...
+
+## Rules
+
+The rules work alongside `proto_library` rules. You can configure `rules_buf` using a [`buf.yaml`][buf_yaml] configuration file. Export the `buf.yaml` using `exports_files(["buf.yaml"])` to reference it. For repositories that contain a [`buf.work.yaml`][buf_work_yaml] that references multiple `buf.yaml` files, you need to export and reference each `buf.yaml` file independently.
+
+> We recommend using the [Gazelle extension](#gazelle) to generate the following rules.
+
+### `buf_dependencies` {#buf-dependencies}
+
+`buf_dependencies` is a [repository rule][repository_rule] that downloads one or more modules from the [BSR][bsr] and generates build files using Gazelle. [Setup Gazelle][gazelle_setup] to use this rule. To also use Gazelle to generate this rule see the [Dependencies](#gazelle-dependencies) section.
```
[the remark this answers is not in the document]
Comment (robbertvanginkel) [reply docs/build-systems/bazel.md:92]: Don't you need our gazelle extension to patch up `deps` automatically in `proto_library` rules?

File: docs/build-systems/bazel.md (hunk 23 above)
Comment (srikrsna-buf) [reply docs/build-systems/bazel.md:68]: Changing this to placeholders

File: docs/build-systems/bazel.md (hunk 26 above)
[the remark this answers is not in the document]
Comment (srikrsna-buf) [reply docs/build-systems/bazel.md:92]: Only if they use gazelle for generating `proto_library` rules

File: docs/build-systems/bazel.md (hunk 24 above)
Comment (srikrsna-buf) [reply docs/build-systems/bazel.md:139]: That's the default one I'll add a note here

**Review by robbertvanginkel — APPROVED** [review]

File: docs/build-systems/bazel.md [hunk 27]
```diff
@@ -0,0 +1,468 @@
...
+
+# imports "validate/validate.proto"
+proto_library(
+    name = "foo_proto",
+    srcs = ["pet.proto"],    
+    deps = ["@buf_deps//validate:validate_proto"],
+)
+```
+
+We recomment using a single `buf_dependencies` rule for each `buf.yaml` file. The [Gazelle extension](#gazelle-dependencies) does this by default.
```
Comment (amckinney) [inline docs/build-systems/bazel.md:126]:
```suggestion
We recommend using a single `buf_dependencies` rule for each `buf.yaml` file. The [Gazelle extension](#gazelle-dependencies) does this by default.
```

File: docs/build-systems/bazel.md [hunk 28]
```diff
@@ -0,0 +1,468 @@
...
+Export the `buf.yaml` file by adding `exports_files(["buf.yaml"])` to the `BUILD` file.
+
+> Inside Buf [workspaces][workspaces], make sure to export each `buf.yaml` file.
+
+Now run Gazelle
+```terminal
+$ bazel run //:gazelle
+```
+
+This will now take care of updating your proto build files, just run `//:gazelle` whenever proto files are added/removed.
```
Comment (amckinney) [inline docs/build-systems/bazel.md:294]:
```suggestion
This will take care of updating your proto build files, just run `//:gazelle` whenever proto files are added/removed.
```

**Review by amckinney — APPROVED** [review]
Nice!
